### PR TITLE
Remove ForcesDefaultVariantIfNull and make it default behavior

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1206058566871760/f

### Description
`ForcesDefaultVariantIfNull` was added with the intent to be used by experiments very close to the app creation.
However after discussing internally (see asana) this should be the default hardcoded behavior. And so the `ForcesDefaultVariantIfNull` annotation is not needed

### Steps to test this PR

_Test_
Repeat tests from https://github.com/duckduckgo/Android/pull/3758

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3943** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3943?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3943?utm_source=stack-comment).

File: feature-toggles/feature-toggles-api/src/main/java/com/duckduckgo/feature/toggles/api/FeatureToggles.kt [hunk 1]
```diff
@@ -231,25 +231,28 @@ internal class ToggleImpl constructor(
...
+                isVariantTreated &&
                 appVersionProvider.invoke() >= (state.minSupportedVersion ?: 0)
         }
         // check if it should always be enabled for internal builds
         if (isInternalAlwaysEnabled && flavorNameProvider.invoke().lowercase() == "internal") {
             return true
         }
         // If there's not assigned variant yet and feature forces default variant, set default variant
-        if (appVariantProvider.invoke() == null && forcesDefaultVariant) {
+        if (appVariantProvider.invoke() == null && isExperiment) {
```
Comment (nalcalag) [inline feature-toggles/feature-toggles-api/src/main/java/com/duckduckgo/feature/toggles/api/FeatureToggles.kt:247]: I think this check would be `!isExperiment`? We want to default variant to null if currentVarianKey is null or if the feature toggle doesn't have the @Experiment annotation

File: feature-toggles/feature-toggles-api/src/main/java/com/duckduckgo/feature/toggles/api/FeatureToggles.kt [hunk 2]
```diff
@@ -231,25 +231,28 @@ internal class ToggleImpl constructor(
...
             return state.enable &&
-                state.isVariantTreated(appVariantProvider.invoke()) &&
+                isVariantTreated &&
                 appVersionProvider.invoke() >= (state.minSupportedVersion ?: 0)
         }
         // check if it should always be enabled for internal builds
         if (isInternalAlwaysEnabled && flavorNameProvider.invoke().lowercase() == "internal") {
             return true
         }
         // If there's not assigned variant yet and feature forces default variant, set default variant
```
Comment (nalcalag) [inline feature-toggles/feature-toggles-api/src/main/java/com/duckduckgo/feature/toggles/api/FeatureToggles.kt:246]: NIT
```suggestion
        // If there's not assigned variant yet and is an experiment feature, set default variant
```

**Review by nalcalag — APPROVED** [review]
Code looks good and works as expected 💯 Thanks for the improvement @aitorvs !

**Pull request merged**